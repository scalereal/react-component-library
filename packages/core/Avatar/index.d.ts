import React from 'react';
import { WithStyle } from '../../utils';
import { AvatarProps } from './types';
export declare const SizeMap: AvatarProps['size'][];
export declare const DisplayMap: AvatarProps['display'][];
declare const Avatar: React.FC<AvatarProps> & WithStyle;
export default Avatar;
