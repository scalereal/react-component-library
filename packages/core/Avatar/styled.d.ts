import { Props } from './types';
export declare const StyledAvatar: import("styled-components").StyledComponent<"div", any, Props, never>;
