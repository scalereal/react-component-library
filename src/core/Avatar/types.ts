
export type AvatarSizes = 'ES' | 'S' | 'M' | 'L' ;

export interface Props{
    size?: AvatarSizes;
    bgColor?:string;
    textColor?:string;
    hoverEffect?:boolean;
    hoverTextColor?:string;
    hoverBgColor?:string;
    hoverShadowColor?:string;
    hoverTextShadowColor?:string;
    res?: string;
    disabled?:boolean;
}