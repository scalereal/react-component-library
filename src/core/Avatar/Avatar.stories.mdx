import Avatar, { DisplayMap, SizeMap } from '.';
import { Preview, Story, Meta, Props, ArgsTable, Canvas } from '@storybook/addon-docs/blocks';
import { withKnobs, color, boolean, select } from '@storybook/addon-knobs';

<Meta
    title="Components/Core/Avatar"
    component={Avatar}
    argTypes={{
        size: {
            control: {
                type: 'select',
                options: Object.values(SizeMap)
            },
            description: 'Avatar Sizes `XXS` | `XS` | `S` | `M` | `L` | `XL`'
        },
        res: {
            control: { type: 'text' },
            description: 'Avatar resource type `string` provide url or name'
        },
        bgColor: {
            control: 'color',
            description: '`string`'
        },
        textColor: {
            control: 'color',
            description: '`string`'
        },
        hoverBgColor: {
            control: 'color',
            description: '`string`'
        },
        hoverTextColor: {
            control: 'color',
            description: '`string`'
        },
        hoverShadowColor: {
            control: 'color',
            description: '`string`'
        },
        hoverTextShadowColor: {
            control: 'color',
            description: '`string`'
        },
        hoverEffect: {
            control: { type: 'boolean' },
            description: '`boolean`'
        },
        disabled: {
            control: { type: 'boolean' },
            description: '`boolean`'
        },
        display: {
            control: {
                type: 'select',
                options: Object.values(DisplayMap)
            },
            description: 'type `string` value `inline-block` | `block` '
        },
        ref: {
            table: {
                disable: true
            }
        },
        theme: {
            table: {
                disable: true
            }
        }
    }}
/>

# Avatar

Displays an image or text that represents a user or organization

export const Template = args => (
    <div>
        <Avatar
            display="inline-block"
            res="https://media.istockphoto.com/photos/child-hands-formig-heart-shape-picture-id951945718?k=6&m=951945718&s=612x612&w=0&h=ih-N7RytxrTfhDyvyTQCA5q5xKoJToKSYgdsJ_mHrv0="
            {...args}
        />
        <Avatar display="inline-block" res="pradip bhusnar" {...args} />
    </div>
);

<Preview>
    <Story name="Avatar" args={{ size: 'S' }}>
        {Template.bind({})}
    </Story>
</Preview>

### Color Variations

You can either change the theme or pass the colors as props.

<Preview withToolbar>
    <div>
        <Avatar
            display="inline-block"
            size="S"
            res="Pradip Bhusnar"
            hoverEffect={true}
            bgColor="green"
            textColor="white"
            hoverBgColor="blue"
        />
        <Avatar
            display="inline-block"
            size="S"
            res="Pradip Bhusnar"
            hoverEffect={true}
            bgColor="red"
            textColor="white"
            hoverBgColor="green"
        />
    </div>
</Preview>

### Sizes

There are Six sizs available `XXS` | `XS` | `S` | `M` | `L` | `XL`.

<Preview withToolbar>
    <div>
        <Avatar display="inline-block" size="XXS" res="Pradip Bhusnar" hoverEffect={true} />
        <Avatar display="inline-block" size="XS" res="Pradip Bhusnar" hoverEffect={true} />
        <Avatar display="inline-block" size="S" res="Pradip Bhusnar" hoverEffect={true} />
        <Avatar display="inline-block" size="M" res="Pradip Bhusnar" hoverEffect={true} />
        <Avatar display="inline-block" size="L" res="Pradip Bhusnar" hoverEffect={true} />
        <Avatar display="inline-block" size="XL" res="Pradip Bhusnar" hoverEffect={true} />
    </div>
</Preview>

### Props

<Props story="Avatar" />
