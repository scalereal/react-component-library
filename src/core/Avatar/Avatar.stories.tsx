import { Meta, Story } from '@storybook/react/types-6-0';
import React from 'react';
import Avatar from '.';
import { Props } from './types';

const SizeMap: Props['size'][] = ['ES', 'S', 'M', 'L'];

export default {
    title: 'Components/Core/Avatar',
    component: Avatar,
    argTypes : {
        size : {
            control: {
                type: 'select',
                options: Object.values(SizeMap),
            },
        },
        bgColor : {
            control:'color'
        },
        textColor: {
            control:'color'
        },
        hoverBgColor: {
            control:'color'
        },
        hoverTextColor: {
            control:'color'
        },
        hoverShadowColor: {
            control:'color'
        },
        hoverTextShadowColor: {
            control:'color'
        },
        hoverEffect:{ 
            control: { type: 'boolean'}  
        },
        disabled:{
            control:{type:'boolean'}
        }
    }
} as Meta;

const Template: Story<Props> = ({...args }) => {
    // retrieves the appropriate icon passes it as a component prop
    return <Avatar {...args} />;
};

export const Default = Template.bind({});
Default.args = {
    res: 'https://media.istockphoto.com/photos/child-hands-formig-heart-shape-picture-id951945718?k=6&m=951945718&s=612x612&w=0&h=ih-N7RytxrTfhDyvyTQCA5q5xKoJToKSYgdsJ_mHrv0=',
};

export const ImageAvatar = Template.bind({});
ImageAvatar.args = {
    res:'pradip bhusnar'
};
