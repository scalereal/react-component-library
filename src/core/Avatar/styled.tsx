import styled from 'styled-components';
import { Props } from './types';


export const StyledAvatar = styled.div<Props>`
    width: 150px;
    height: 150px;
    line-height: 150px;
    border-radius: 50%;
    text-align:center;
    background-color: ${pr => pr.bgColor || `blue` };

    ${pr => pr.hoverEffect? `&:hover {
        background-color: ${pr.hoverBgColor};
        span{
            color: ${pr.hoverTextColor};
            text-shadow: 0px 5px 8px ${pr.hoverTextShadowColor };
        }
        img{
            box-shadow: 0px 5px 8px ${pr.hoverImageShadowColor };
        }
        }
        `:
        ''
    }

    img{
        width: 150px;
        height: 150px;
        border-radius: 50%;
    }
    
    span{
        display: inline-block;
        vertical-align: middle;
        line-height: normal;
        font-size:5em;
        color: ${pr => pr.textColor };
    }
`;