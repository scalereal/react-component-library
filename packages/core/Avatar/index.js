var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import React from 'react';
import { StyledAvatar } from './styled';
export var SizeMap = ['XXS', 'XS', 'S', 'M', 'L', 'XL'];
export var DisplayMap = ['inline-block', 'block'];
var Avatar = React.memo(React.forwardRef(function (props, ref) {
    var size = props.size, _a = props.res, res = _a === void 0 ? 'Pradip Bhusnar' : _a;
    var isValidURL = function (res) {
        var reso = res.match(/(http(s)?:\/\/.)?(www\.)?[-a-zA-Z0-9@:%._\\+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_\\+.~#?&//=]*)/g);
        return (reso !== null);
    };
    var chars = function (name) {
        var nameSplit = String(name).toUpperCase().split(' ');
        if (nameSplit.length == 1)
            return nameSplit[0].charAt(0);
        return nameSplit[0].charAt(0) + nameSplit[1].charAt(0);
    };
    return (React.createElement(StyledAvatar, __assign({ ref: ref, size: size }, props), isValidURL(res) ? React.createElement("img", { src: res }) : React.createElement("span", null, chars(res))));
}));
Avatar.displayName = "Avatar";
Avatar.defaultProps = {
    size: 'S',
    hoverEffect: true
};
export default Avatar;
