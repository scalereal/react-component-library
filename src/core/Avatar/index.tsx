import * as React from 'react';
import { WithStyle } from '../../utils';
import { StyledAvatar } from './styled';
import { AvatarProps } from './types';

export const SizeMap: AvatarProps['size'][] = ['XXS','XS', 'S', 'M', 'L', 'XL'];
export const DisplayMap: AvatarProps['display'][] = ['inline-block','block']

const Avatar:React.FC<AvatarProps> & WithStyle = React.memo(React.forwardRef( (props, ref) => {
    const {size,res='Pradip Bhusnar'} = props;

    const isValidURL = (res:string) => {
        const reso = res.match(/(http(s)?:\/\/.)?(www\.)?[-a-zA-Z0-9@:%._\\+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_\\+.~#?&//=]*)/g);
        return (reso !== null)
    };
      
    const chars =(name:string)=>{
        const nameSplit = String(name).toUpperCase().split(' ');
        if (nameSplit.length == 1) return nameSplit[0].charAt(0);
        return nameSplit[0].charAt(0) + nameSplit[1].charAt(0);
    }
    

    return (
        <StyledAvatar ref={ref} size={size} {...props}>
            {isValidURL(res)?<img src={res} />:<span>{chars(res)}</span>}
        </StyledAvatar>
    );

}));

Avatar.displayName="Avatar";
Avatar.defaultProps = {
    size:'S',
    hoverEffect: true
};

export default Avatar;
