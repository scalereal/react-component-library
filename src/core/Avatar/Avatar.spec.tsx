import { render } from '@testing-library/react';
import * as React from 'react';
import { Avatar } from './Avatar';

describe('Avatar component', () => {
    it('should render with default theme', () => {
        const { container } = render(<Avatar/>);
        expect(container).toMatchSnapshot();
    });
});