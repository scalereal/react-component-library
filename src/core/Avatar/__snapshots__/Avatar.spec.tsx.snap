// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar component should render image avatar properly 1`] = `
<div>
  <div
    class="sc-bdfBwQ kTamvn"
  >
    <img
      src="https://media.istockphoto.com/photos/child-hands-formig-heart-shape-picture-id951945718?k=6&m=951945718&s=612x612&w=0&h=ih-N7RytxrTfhDyvyTQCA5q5xKoJToKSYgdsJ_mHrv0="
    />
  </div>
</div>
`;

exports[`Avatar component should render with all the props given 1`] = `
<div>
  <div
    class="sc-bdfBwQ jERgWU"
  >
    <span>
      PB
    </span>
  </div>
</div>
`;

exports[`Avatar component should render with default theme 1`] = `
<div>
  <div
    class="sc-bdfBwQ cXhJVx"
  >
    <span>
      PB
    </span>
  </div>
</div>
`;

exports[`Avatar component should render with not-allowed cursor when it is disabled 1`] = `
<div>
  <div
    class="sc-bdfBwQ dTrERv"
    disabled=""
  >
    <span>
      PB
    </span>
  </div>
</div>
`;
